modules/platform/nuxeo-automation/nuxeo-automation-features/src/main/java/org/nuxeo/ecm/automation/core/operations/coldstorage/MoveToColdStorage.java

```diff
@@ -50,8 +55,28 @@
     @Param(name = "save", required = false, values = "true")
     protected boolean save = true;
 
-    @OperationMethod(collector = DocumentModelCollector.class)
-    public DocumentModel run(DocumentModel doc) {
+    @OperationMethod
+    public DocumentModelList run(DocumentModelList documents) {
+        DocumentModelList result = new DocumentModelListImpl();
+        for (DocumentModel documentModel : documents) {
+            try {
+                result.add(moveContent(documentModel, false));
+            } catch (Exception e) {
```

Comment:
Why catching `Exception` here? Do not forget https://doc.nuxeo.com/corg/catching-exceptions/#never-catch-exception :)

Reply:
I completely agree and i am not in favour of catch `Exception` level, I was going to cache `NuxeoException` but the move of document**s** can be see as a batch process (i asked on slack channel) and the idea is to be able to continue the move even one document fails it is the reason why i ended by catching `Exception`

Reply:
@troger , what is usually the approach for this kind of use cases?
I can see that we have a some operations that just loop the desired method which means that it will stop at the first failure. Should be this the pattern we want?

Reply:
If you have to deal with catching `Exception` then follow the documentation to deal with `InterruptedException`: https://doc.nuxeo.com/corg/catching-exceptions/#methods-that-want-to-keep-running-after-an-exception

Reply:
Anyway, I still think that you need to just catch `NuxeoException`, so you continue if there is an issue in Nuxeo related code. Other runtime exceptions should just break the code anyway... wdyt @efge?

Reply:
yes i had a look at the document, `InterruptedException` is type of Exception and not a runtime one, with catch it will make an error compilation. Which means i should add the `InterruptedException` into the `public DocumentModel run(DocumentModel document) throws InterruptedException` if we agree i do the modification.

Reply:
catching `NuxeoException` is another solution why not, my wondering was about any other affect to avoid blocking the whole move. but if we all agree let add the `NuxeoException` only.

Reply:
I would go for `NuxeoException` only, other runtime exception must break.

Reply:
ok I agree it is what i did yesterday